src/librbd/AioImageRequest.cc

```diff
@@ -462,8 +462,15 @@ void AioImageFlush::send_request() {
     m_aio_comp->associate_journal_event(journal_tid);
   }
 
-  C_AioRequest *req_comp = new C_AioRequest(cct, m_aio_comp);
-  m_image_ctx.flush(req_comp);
+  if (!journaling && m_image_ctx.object_cacher != NULL)
```

Comment:
I see, would it be 0 in some case? e.g, `journaling == false && m_image_ctx.object_cacher == NULL`